### ISSUE TITLE
Dirver test.

### ISSUE THREAD

**Issue by giulioungaretti** [QCoDeS/Qcodes#475]
We need some solution (that should have been baked into the base class to begin with) to make sure the drivers are actually valid python.

**Comment by jenshnielsen** [comment]
I.e #474 
and 
in #459 the default values could not validate

**Comment by giulioungaretti** [comment]
@jenshnielsen  I am thinking about having a mock, that simply replaces base, and tests if all the instruments at least can be instantiated.

**Comment by jenshnielsen** [comment]
Yes agreed, have been doing that manually when testing drivers for instruments that I don't have easy access to. Above is just some examples of why we need it

**Comment by sohailc** [comment]
Making a mock instrument server which receives messages and sends reply's on specified ports on localhost is I think the best option for making being able to run and debug instrument drivers, at least for IP instruments.  In this way we can leave the driver unmodified, that is, the driver being tested is exactly the same as the driver that will actually be used. I have made mock instrument server in my PR (https://github.com/QCoDeS/Qcodes/pull/700), but maybe it is a good idea to open a seperate PR specifically for this problem.

**Comment by WilliamHPNielsen** [comment]
I guess this was dealt with by #859. At least for VISA instruments there is now a standard way of doing this.
